### DISCUSSION TITLE
Adding Lambda as a key in logs

### DISCUSSION THREAD

**Discussion by Padwicker** [aws-powertools/powertools-lambda-python#367] [Q&A] [answered]
I need to add lambda as a key in my structured logs, I've tried using the following syntax but it doesn't work because lambda is a reserved word in python

`logger.structure_logs(append=True, lambda="whatever")`

Is there another way I can achieve this using the powertools logger?

**Comment by deric4** [comment, marked as answer]
I haven't tried it specifically for this library, but you might be able to get away with [implementing dictionary unpacking ](https://medium.com/swlh/how-to-pack-and-unpack-data-in-python-tuples-and-dictionaries-55d218c65674)which might look something like this :

```python
fields_to_append = {'lambda': 'whatever'}
logger.structure_logs(append=True, **fields_to_append)
```

or, theres always using a different field name lol 🙃

**Comment by heitorlessa** [comment]
Hey, thanks for opening a discussion ;)

Sure, try this and let us know if it works, as this is a problem for any reserved keyword for function kwarg:

logger.structure_logs(append=True, **{"lambda": "whatever"})

I'd typically ask what the use case is as together we might come up with a better name, but I'd leave at your discretion.

Thanks

**Reply by Padwicker** [reply]
Thanks for the suggestions :) I'll get back to you with results. I just need to use that key to get our logging to work nicely with datadog.

**Comment by heitorlessa** [comment]
Sure thing :) Both suggestions work as it's a common solution for not only
using reserved keyword as keyword arguments but also to long function
parameter calls when you use a dic - Just tested with Logger to triple
check.

from aws_lambda_powertools import Logger

logger = Logger(service="discussion_367")
# literal form
logger.structure_logs(append=True, **{"lambda": "test"})
# from a dict
keys_to_append = {"lambda": "test"}
logger.structure_logs(append=True, **keys_to_append)

logger.info("hello")


```json
{"level": "INFO", "location": "<module>:6", "message": "hello",
"timestamp": "2021-03-26 21:02:22,477", "service": "discussion_367",
"sampling_rate": 0.0, "lambda": "test"}
```

**Reply by Padwicker** [reply]
Thanks, btw love this library, it's so useful
